spikeinterface/sortingcomponents/dimensionality_reduction.py

```diff
@@ -0,0 +1,81 @@
+import pickle
+
+import numpy as np
+
+from spikeinterface.sortingcomponents.peak_pipeline import run_peak_pipeline, PeakPipelineStep
+from spikeinterface.sortingcomponents.peak_detection import detect_peaks
+from spikeinterface.sortingcomponents.peak_selection import select_peaks
+from spikeinterface.postprocessing import compute_principal_components
+from spikeinterface.core.sparsity import ChannelSparsity
+from spikeinterface import extract_waveforms, NumpySorting
+
+
+class TemporalPCA(PeakPipelineStep):
+    need_waveforms = True
```

Comment:
Can you add docstrings to the class too?

Reply:
Done.